devices.js

```diff
@@ -11735,7 +11735,23 @@ const devices = [
         model: 'SR-ZG9040A',
         vendor: 'Sunricher',
         description: 'Zigbee micro smart dimmer',
-        extend: preset.light_onoff_brightness(),
+        fromZigbee: preset.light_onoff_brightness().fromZigbee.concat([fz.electrical_measurement, fz.metering, fz.ignore_genOta]),
+        toZigbee: preset.light_onoff_brightness().toZigbee,
+        meta: {configureKey: 2},
+        exposes: [e.light_brightness(), e.power(), e.energy()],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            const endpoint = device.getEndpoint(1);
+            const binds = ['genOnOff', 'genLevelCtrl', 'haElectricalMeasurement', 'seMetering'];
+            await reporting.bind(endpoint, coordinatorEndpoint, binds);
+            await reporting.onOff(endpoint);
+            await reporting.brightness(endpoint);
+            await reporting.readEletricalMeasurementMultiplierDivisors(endpoint);
+            await reporting.activePower(endpoint);
+            await reporting.rmsCurrent(endpoint, {min: 10, change: 10});
+            await reporting.rmsVoltage(endpoint, {min: 10});
```

Comment:
Reporting is setup for voltage and current here but it's missing from `exposes`. Does this device provide voltage and current measurements? If yes add it to exposes, if not remove the setup reporting lines.

Reply:
yes, `e.voltage(), e.current(),` should be added. I'm realizing that `SR-ZG9040A` and `ZG2835RAC` (also missing to expose voltage and current) is exactly the same config. Should they be two different setup with only different `zigbeeModel:`, `model:` and `description:`, or should this be refactored some way?

Reply:
You can add the same code there